libs/datasets/beds.py

```diff
@@ -1,56 +1,90 @@
 from libs.datasets.dataset_utils import AggregationLevel
 from libs.datasets import custom_aggregations
 
 
 class BedsDataset(object):
     class Fields(object):
+        COUNTRY = "country"
         STATE = "state"
         FIPS = "fips"
         STAFFED_BEDS = "staffed_beds"
         LICENSED_BEDS = "licensed_beds"
         ICU_BEDS = "icu_beds"
         AGGREGATE_LEVEL = "aggregate_level"
 
-        # Fields generated in `from_source`
         COUNTY = "county"
         SOURCE = "source"
         GENERATED = "generated"
+        # Combined beds are the max of staffed and licensed beds.
+        # The need for this may be dataset specific, but there are cases in the
+        # DH Beds dataset where a source only has either staffed or licensed.
+        # To provide a better best guess, we take the max of the two in `from_source`.
+        MAX_BED_COUNT = "max_bed_count"
+
+    # Key for grouping data at the state level.
+    # This includes other keys that should be preserved - for example, this will preserve
+    # the source field during a groupby.
+    STATE_GROUP_KEY = [
+        Fields.SOURCE,
+        Fields.AGGREGATE_LEVEL,
+        Fields.GENERATED,
+        Fields.COUNTRY,
+        Fields.STATE,
+    ]
 
     def __init__(self, data):
         self.data = data
 
+    @property
+    def state_data(self) -> pd.DataFrame:
+        """Returns a new BedsDataset containing only state data."""
+        is_state = (
+            self.data[self.Fields.AGGREGATE_LEVEL] == AggregationLevel.STATE.value
+        )
+        return self.data[is_state]
+
+    @property
+    def county_data(self) -> pd.DataFrame:
+        """Returns a new BedsDataset containing only county data."""
+        is_county = (
+            self.data[self.Fields.AGGREGATE_LEVEL] == AggregationLevel.COUNTY.value
+        )
+        return self.data[is_county]
+
     @classmethod
     def from_source(cls, source: "DataSource", fill_missing_state=True):
         """Loads data from a specific datasource.
 
         Remaps columns from source dataset, fills in missing data
         by computing aggregates, and adds standardized county names from fips.
+
+        Args:
+            source: Data source.
+            fill_missing_state: If True, fills in missing state level data by
+                aggregating county level for a given state.
         """
         if not source.BEDS_FIELD_MAP:
             raise ValueError("Source must have beds field map.")
 
         data = source.data
+
         to_common_fields = {value: key for key, value in source.BEDS_FIELD_MAP.items()}
         final_columns = to_common_fields.values()
         data = data.rename(columns=to_common_fields)[final_columns]
         data[cls.Fields.SOURCE] = source.SOURCE_NAME
         data[cls.Fields.GENERATED] = False
 
-        group = [cls.Fields.SOURCE, cls.Fields.AGGREGATE_LEVEL, cls.Fields.STATE, cls.Fields.GENERATED]
-        data = custom_aggregations.update_with_combined_new_york_counties(
```

Comment:
The reason we don't need this aggregation any more is because we're only operating at the state level? If so, I think we can remove the import.